### ISSUE TITLE
Authentication failed (407) Proxy authentication required

### ISSUE THREAD

**Issue by ph3x** [Kyrodan/KeeAnywhere#85]
Hi,

I was waiting for the new version since I also got auth failed as in
#74. But maybe this is a new issue - log is attached.
I´m trying to get this plugin working on my office PC which is of
course behind a company proxy.

If I can provide any more informations I´ll be happy to do so.

Cheers,
Christian

Authentication failed!

Fehler beim Senden der Anforderung.
Google.Apis.Core
   bei Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__55.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Google.Apis.Auth.OAuth2.Requests.TokenRequestExtenstions.<ExecuteAsync>d__0.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Google.Apis.Auth.OAuth2.Flows.AuthorizationCodeFlow.<FetchTokenAsync>d__35.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Google.Apis.Auth.OAuth2.Flows.AuthorizationCodeFlow.<ExchangeCodeForTokenAsync>d__30.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei KeeAnywhere.StorageProviders.GoogleDrive.GoogleDriveStorageConfigurator.<Claim>d__c.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei KeeAnywhere.OAuth2.OAuth2Form.<OnNavigated>d__3.MoveNext()
Void MoveNext()

Inner:
Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich.
System
   bei System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   bei System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
System.IO.Stream EndGetRequestStream(System.IAsyncResult, System.Net.TransportContext ByRef)

**Comment by Kyrodan** [comment]
Hi @ph3x ,

is proxy setting in *KeePass* configured correctly? 
See details here: https://github.com/Kyrodan/KeeAnywhere/wiki/Getting-Started#using-a-proxy

If yes: 
* does "Help - Check for Updates" reports a proper result?
* can you check, wheter another Cloud Provider works (e.g. Dropbox)
* provide some information of your type of proxy

Regards,
Daniel

**Comment by ph3x** [comment]
I tried all possible settings in the proxy window but I´m always getting the "proxy authentication required" error. Also the update dialog shows an error for KeePass "Update check failed. Version information file cannot be downloaded." - however KeeAnywhere is showing "Up to date".

As far as I know we are using Cisco Iron Port Appliances but I don´t think I will get any more informations
out of our IT/security team.

I know that M$ Internet Explorer and Google Chrome are only working if proxy settings are enabled -
and therefore some sites are blocked on proxy level. However we are able to use Firefox without
proxy settings enabled and are able to open all sites. Even a portable version of Google Drive is working
with proxy set or direct connection.

**Comment by Kyrodan** [comment]
If update info for KeeAnywhere can be fetched successfully and for KeePass not and you tried all proxy settings: then I assume that I cannot do anything for you. It seems, that your corporate network blocks some sites and some not, maybe especially SSL-Encrypted ones (https://).

Can you download https://sslsites.de/keepass.info/update/version2x.txt.gz with IE or FF?
Can you open https://github.com/ in IE or FF?

**Comment by ph3x** [comment]
Both links can be openend in IE, FF and Chrome.
Even http://keepass.info/update/version2x.txt.gz can be opened in all browsers.

**Comment by Kyrodan** [comment]
hmm, interesting. But as I mentioned: if you tried all settings and with none of them the update check could be performed successfully and completely, I cannot help. I'm sorry :-(
In my opinion, only your IT knows the correct settings.

**Comment by ph3x** [comment]
Stupid me - or stupid KeePass. Looks like one has to restart KeePass to have it using the proxy settings.
Right now update check and connection to google drive is wokring. I´ll have a look at the sync later this evening when I´m at home

**Comment by ph3x** [comment]
Ok, looks like I have one more problem to solve before I can test the sync...
If I try to save a (new) database to my google drive I get the following error presented:

![image](https://cloud.githubusercontent.com/assets/9625650/26630871/e7036582-4608-11e7-9c8b-7e34f5d7a4e6.png)

In KeePass Advanced settings I have checked Verify written file after saving a database
and Use file transactions for writing databases.

**Comment by Kyrodan** [comment]
Glad to hear the first problem is solved :-) 

I can't reproduce this new issue in my first try. Both of your settings are valid and should work.

Do you have one of the new features (Auto Backup or Offline Caching) enabled? If yes, please switch off and try again. Otherwise: please start KeePass with parameter "-debug" and make a screenshot of the (hopefully more precise) error message.

**Comment by ph3x** [comment]
In fact I had enabled Auto Backup but I also get the same error when it is disabled.
Even with -debug I get the same popup without any more useful information.
The only thing I could probably imagine leading to this error (I´m no developer though) is
that there is a space in the file path like gdrive:///This Location/test.kdbx

**Comment by Kyrodan** [comment]
Ok, just to make sure, what you did:
1. Started KeePass
2. Created a new Database -a (local) save dialog appears: you entered a filename
3. Choose "Save" / "Save to Cloud Drive"
4. Select folder and filename in File Save Dialog
5. Click "OK"
6. Messagebox with error appears

Is that right?
What happens, if you add a new entry in this (empty) database and then try to save it?
Can you open an existing database from your Google Drive, change something and then save it?

**Comment by ph3x** [comment]
Daniel,

I think I had the chance to narrow down this issue - will see if it works in the office tomorrow too...

What works:
*) Copy a database in the main folder (which is openend when selecting "Open from Cloud Drive"/"Save to Cloud Drive" - gdrive:///User Name/database.kdbx. Fire up KeePass, Open from Cloud Drive, edit database, Save (Ctrl + S)
*) Open a database (or create a new one) and save it with "Save to Cloud Drive" in a folder below root - e.g. gdrive:///User Name/Test/database.kdbx. Also editing and saving works here.

What does not work:
*) Create a new database, edit it and save it with "Save to Cloud Drive" in gdrive:///User Name/database.kdbx
*) Open an existing database, edit it and save it with "Save to Cloud Drive" in gdrive:///User Name/database.kdbx

I can add screenshots if something is not clear and I will report tomorrow how things are going on the office PC but I think it will work there too when not using the folder gdrive:///User Name

Cheers,
Christian

**Comment by ph3x** [comment]
Good morning Daniel,

I can confirm it´s working also on the office PC with the same limitations as
mentioned above.

Cheers,
Christian

**Comment by Kyrodan** [comment]
Hi @ph3x ,

I can reproduce your latest problem. I have created a new issue #88 for that and close this now.
